services/wireguard/resources/allocator_unix.go

```diff
@@ -126,7 +124,12 @@ func (a *Allocator) AllocatePort() (int, error) {
 	a.mu.Lock()
 	defer a.mu.Unlock()
 
-	for i := a.portMin; i < a.portMax; i++ {
+	if err := a.initializePortMin(); err != nil {
+		return 0, err
+	}
+
+	portMax := a.portMin + a.maxConnections
```

Comment:
Can it go out of the available pool?
let say we have 
```
		start:    40000,
		capacity: 10000,
```
We acquired 49999 port, and adding 255 to it will be out of the range.

Reply:
Same as https://github.com/mysteriumnetwork/node/pull/880#discussion_r275735324